hkmc2/shared/src/main/scala/hkmc2/utils/Scope.scala

```diff
@@ -53,6 +53,8 @@ class Scope
       else thisError(thisSym)
     )
     curThis match
+    case S(S(sym: TopLevelSymbol)) if sym === State.globalThisSymbol =>
+      "globalThis" // `this` refers to `undefined` in strict mode
```

Comment:
```suggestion
      // `this` at the top level evaluates to `undefined` in strict mode.
      // We need this because generate some code that uses `this`/`globalThis`,
      // for example the symbol loading code.
      "globalThis"
```